qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/20_camel_case_on_format.yml

```diff
@@ -0,0 +1,79 @@
+---
+"Create index with camel case on format (allowed with warning in 7.x)":
+  - skip:
+      version: "8.0.0 - "
+      reason:  "at version 8.0.0, camel case is not allowed"
+      features: "warnings"
+  - do:
+      warnings:
+        - "Camel case format name strictDateOptionalTime is deprecated and will be removed in a future version. Use snake case name strict_date_optional_time instead."
```

Comment:
I was curious where we issue this deprecation warning on index creation, but I couldn't find it in the PR 🤔

Reply:
I think anytime a transport action uses the DeprecationLogger, it will append to disk and also set the header warning using the HeaderWarningAppender. So in this case, once something like TransportPutMappingAction matches the deprecation in the deprecation checks, while inside logDeprecation, it will print exactly what we put as text inside DeprecationChecks. So all we had to do is bring back `IndexDeprecationChecks.deprecatedCamelCasePattern(indexMetadata)` in the INDEX_SETTINGS_CHECKS.

Reply:
Okay, I think I figured out what's happening. Adding something `IndexDeprecationChecks` doesn't affect what warning headers are returned from index creation/ update API calls (to the best of my knowledge) -- it just surfaces the warning in the deprecation info API. The warning in this test is coming from the 7.17 cluster (which explains why I couldn't find it in the current code!)

In summary, we will only warn in 7.17 when the index is created, and also surface a warning in the deprecation info API. All other API calls in 8.x won't give a warning. This seems okay to me.